extensions.bzl

```
@@ -0,0 +1,71 @@
+load("//:rules.bzl", bb_macro = "buildbuddy")
+
+def _ext_impl(mctx):
+    found_root_modules = False
+    root_modules = struct()
+    for m in mctx.modules:
+        if m.is_root:
+            found_root_modules = True
+            root_modules = m
+            break
+    if not found_root_modules:
+        fail("buildbuddy must be used in root module")
+
+    if len(root_modules.tags.platform) > 1:
+        fail("buildbuddy.platform can only be specified once per module")
+    if len(root_modules.tags.gcc_toolchain) > 1:
+        fail("buildbuddy.gcc_toolchain can only be specified once per module")
+    if len(root_modules.tags.msvc_toolchain) > 1:
+        fail("buildbuddy.msvc_toolchain can only be specified once per module")
```

```suggestion
    if len(root_modules.tags.platform) > 1:
        fail("buildbuddy.platform can only be specified once per module")
    platform = root_modules.tags.platform[0]
    if len(root_modules.tags.gcc_toolchain) > 1:
        fail("buildbuddy.gcc_toolchain can only be specified once per module")
    gcc_toolchain = root_modules.tags.gcc_toolchain[0]
    if len(root_modules.tags.msvc_toolchain) > 1:
        fail("buildbuddy.msvc_toolchain can only be specified once per module")
    msvc_toolchain = root_modules.tags.msvc_toolchain[0]
```

avoids having to deal with arrays below

---

Can't do this because the tags could be zero length by default

```
ERROR: /private/var/tmp/_bazel_sluongng/801123c357b08df08cd18dfb8815f909/external/buildbuddy_toolchain~/extensions.bzl:16:45: Traceback (most recent call last):
	File "/private/var/tmp/_bazel_sluongng/801123c357b08df08cd18dfb8815f909/external/buildbuddy_toolchain~/extensions.bzl", line 16, column 45, in _ext_impl
		platform_tag = root_module.tags.platform[0]
Error: index out of range (index is 0, but sequence has 0 elements)
```